### ISSUE TITLE
amq 7 broker does not have any addresses or topics configured

### ISSUE THREAD

**Issue by jirkadanek** [RHsyseng/IntegrationApp-Automation#32]
In amq broker documentation it is said (somewhere), that using the autocreate feature in production deployments is an antipattern.

All queues and topics used should be created in broker.xml configuration, which means that in the openshift broker deployment template, the `alert` topic needs to be created.

**Comment by redhatHameed** [comment]
@jdanekrh if I understand correctly you mean mentioned the names here - https://github.com/jboss-container-images/jboss-amq-7-broker-openshift-image/blob/16108bf61f8d5403e11e412843259797410e99be/templates/amq-broker-74-basic.yaml#L191-L195
but this is only applicable for Address and Queues names.

**Comment by jirkadanek** [comment]
Yes, that's what I meant. @rvais how do I declare a topic in the broker config? There is AMQ_QUEUES, but not AMQ_TOPIC there!

**Comment by rvais** [comment]
There is no "TOPIC" thing per se. You set the queue as multicast type queue. I am not sure the templates support it at the moment.

**Comment by rvais** [comment]
So I take my previous comment back ... AMQ_QUEUE creates ANYCAST and AMQ_ADDRESS creates MULTICAST address ... However there is currently a bug which makes it so the address doesn't have assigned the routing type. https://issues.jboss.org/browse/ENTMQBR-2413
